nrf_security/Kconfig.legacy

```diff
@@ -76,81 +76,102 @@ config MBEDTLS_AES_DECRYPT_ALT
 
 config MBEDTLS_AES_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || (PSA_CRYPTO_DRIVER_CC3XX || \
+		   PSA_CRYPTO_DRIVER_HAS_CIPHER_SUPPORT_CC3XX)
 	default y
 
 config MBEDTLS_CMAC_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || (PSA_CRYPTO_DRIVER_CC3XX || \
+		   PSA_CRYPTO_DRIVER_ALG_CMAC_CC3XX)
 	default y
 
 config MBEDTLS_CCM_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || (PSA_CRYPTO_DRIVER_CC3XX || \
+		   PSA_CRYPTO_DRIVER_ALG_CCM_CC3XX)
 	default y
 
 config MBEDTLS_GCM_ALT
 	bool
 	depends on CC312_BACKEND || \
-		   (PSA_CRYPTO_DRIVER_CC3XX && CRYPTOCELL_CC312_USABLE)
+		   (PSA_CRYPTO_DRIVER_CC3XX && CRYPTOCELL_CC312_USABLE && \
+		   PSA_CRYPTO_DRIVER_ALG_GCM_CC3XX)
 	default y
 
 config MBEDTLS_CHACHA20_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || (PSA_CRYPTO_DRIVER_OBERON    && \
+		   PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_OBERON)|| \
+		   CC3XX_BACKEND || (PSA_CRYPTO_DRIVER_CC3XX      && \
+		   PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_CC3XX)
 	default y
 
 config MBEDTLS_POLY1305_ALT
 	bool
-	depends on OBERON_BACKEND || PSA_CRYPTO_DRIVER_OBERON || \
-		   CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on OBERON_BACKEND || (PSA_CRYPTO_DRIVER_OBERON    && \
+		   PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_OBERON)|| \
+		   CC3XX_BACKEND || (PSA_CRYPTO_DRIVER_CC3XX      && \
+		   PSA_CRYPTO_DRIVER_ALG_CHACHA20_POLY1305_CC3XX)
 	default y
 
 config MBEDTLS_CHACHAPOLY_ALT
 	bool
-	depends on CC3XX_BACKEND || PSA_CRYPTO_DRIVER_CC3XX
+	depends on CC3XX_BACKEND || (PSA_CRYPTO_DRIVER_CC3XX && \
```

Comment:
As a simplification: All PSA_CRYPTO_x_CC3XX symbols depend on PSA_CRYPTO_DRIVER_CC3XX.
So you could simplify these && statements to only check PSA_CRYPTO_x_CC3XX feature instead.

Reply:
Simplification has merit, will be changed to that